monitor/query.go

```diff
@@ -166,15 +182,18 @@ func (m *Monitor) queryReportedCheckpointBTCHeightWithRetry(hashStr string) (*mo
 func (m *Monitor) queryBTCHeaderChainTipWithRetry() (*btclctypes.QueryTipResponse, error) {
 	var btcHeaderChainTipRes btclctypes.QueryTipResponse
 
-	if err := retry.Do(m.ComCfg.RetrySleepTime, m.ComCfg.MaxRetrySleepTime, func() error {
+	if err := retry.Do(func() error {
 		res, err := m.BBNQuerier.BTCHeaderChainTip()
 		if err != nil {
 			return err
 		}
 
 		btcHeaderChainTipRes = *res
 		return nil
-	}); err != nil {
+	},
+		retry.Delay(m.ComCfg.RetrySleepTime),
+		retry.MaxDelay(m.ComCfg.MaxRetrySleepTime),
```

Comment:
I would say we need to pass attempts config to every method, it would pretty wierd that some methods do not respect config value.